sdk/core/Azure.Core/src/Shared/NextLinkOperationImplementation.cs

```diff
@@ -15,20 +18,24 @@ namespace Azure.Core
 {
     internal class NextLinkOperationImplementation : IOperation
     {
+        internal const string RehydrationTokenVersion = "1.0.0";
         private const string ApiVersionParam = "api-version";
         private static readonly string[] FailureStates = { "failed", "canceled" };
         private static readonly string[] SuccessStates = { "succeeded" };
 
         private readonly HeaderSource _headerSource;
         private readonly Uri _startRequestUri;
         private readonly OperationFinalStateVia _finalStateVia;
-        private readonly RequestMethod _requestMethod;
         private readonly HttpPipeline _pipeline;
         private readonly string? _apiVersion;
 
         private string? _lastKnownLocation;
         private string _nextRequestUri;
 
+        // We can only get OperationId if the operation is in progress when the nextRequestUri contains it
+        public string? OperationId { get; private set; }
```

Comment:
We have to make OperationId nullable, because there are cases we can't get it.
For instance, a PUT operation is complete and the nextRequestUri is a get Uri of the resource, we can't get the OperationId from nextRequestUri.

The only case we can get operation id is, when the operation is in progress and we update nextRequestUri, we get OperationId at the same time [here](https://github.com/Azure/azure-sdk-for-net/pull/42686/files#diff-ddf89b49baf148b40544d15990fb244dfa02c26a8a67da24903cb6af0b023e1cR272)

Reply:
Make `OperationId` not nullable and set it default to "NOT_SET"